File: csharp/bird-watcher/BirdWatcher.cs
using System;

class BirdCount
{
    private int[] birdsPerDay;

    public BirdCount(int[] birdsPerDay)
    {
        this.birdsPerDay = birdsPerDay;
    }

    public static int[] LastWeek()
    {
        return new int[] { 0, 2, 5, 3, 7, 8, 4 };
    }

    public int Today()
    {
        return birdsPerDay[^1]; //In C# 8 there is a new operator for indexing: char[] arr = {'c', 'b', 'a'};   int a_last = arr[^1];   int b_second_last = arr[^2];
    }

    public void IncrementTodaysCount()
    {
        birdsPerDay[^1] += 1;
    }

    public bool HasDayWithoutBirds()
    {
        foreach (int number in birdsPerDay)
        {
            if (number == 0)
            {
                return true;
                break;
            }
        }
        return false;
    }

    public int CountForFirstDays(int numberOfDays)
    {
        int total = 0;

        for (int i = 0; i < numberOfDays; i++)
        {
            total += birdsPerDay[i];
        }
        return total;
    }

    public int BusyDays()
    {
        int counter = 0;

        foreach (int number in birdsPerDay)
        {
            if (number >= 5)
            {
                counter++;
            }
        }
        return counter;
    }
}

File: java/bird-watcher/src/main/java/BirdWatcher.java

class BirdWatcher {
    private final int[] birdsPerDay;

    public BirdWatcher(int[] birdsPerDay) {
        this.birdsPerDay = birdsPerDay.clone();
    }

    public int[] getLastWeek() {
        return birdsPerDay; 
    }

    public int getToday() {
        if (birdsPerDay.length == 0){
            return 0;
        }
        return birdsPerDay[6];
    }

    public void incrementTodaysCount() {
        birdsPerDay[6] = birdsPerDay[6] + 1;
    }

    public boolean hasDayWithoutBirds() {
        for (int birdsPerDay:birdsPerDay) {
            if (birdsPerDay == 0){
                return true;
            }
        }
        return false;
    }

    public int getCountForFirstDays(int numberOfDays) {
        int sum = 0;
        if (numberOfDays <= birdsPerDay.length){
           for (int index = 0; index < numberOfDays; ++index) {
            sum += birdsPerDay[index];
        }
        return sum;
        }
        return getCountForFirstDays(7);
        
    }

    public int getBusyDays() {
        int count = 0;
        for (int birdsPerDay:birdsPerDay) {
            if (birdsPerDay >= 5){
                count++;
            }
        }
        return count;
    }
}
